### DISCUSSION TITLE
Run glasskube serve on server side

### DISCUSSION THREAD

**Discussion by Honigeintopf** [glasskube/glasskube#1043] [Q&A] [answered]
Hello,

I appreciate the capabilities of Glasskube, but I find it a bit unintuitive to have to use glasskube serve to start a local server on my machine for accessing the UI. I would prefer to have Glasskube deployed directly within my managed Kubernetes cluster so that the UI is consistently available.

Is there an alternative approach or a way to deploy Glasskube as a service within the cluster, ensuring the UI is always accessible without needing to run it locally?

Thank you!

**Comment by christophenne** [comment, marked as answer]
Hello, 

thanks for raising this question. 

**TL;DR** There is a [`--host` flag](https://github.com/glasskube/glasskube/pull/1027) which you could try. We can't really recommend this setup yet though – for example there is already a follow-up issue: https://github.com/glasskube/glasskube/issues/1039

True, running the UI server locally might seem a bit unintuitive, but it's more of a pragmatic choice, that I think has served us well so far. We initially started it this way because it was way easier to get going, and because we get some basic authorization out of the box with the local kubeconfig. If we put the server into the cluster, we lose that, and anybody that has access to that service can automatically manage the whole cluster. To avoid that, we would need an authentication/authorization layer for the UI, and we didn't want to do this in the even earlier stages (because priorities). So far we have not seen a very convincing reason to step away from this approach. Which of course doesn't mean it will never be done. 
However, we would rather see where Glasskube goes in the next weeks/months – our main priority is the package management aspect, with the UI just being one of multiple ways to interact with it. And as you might know from experience, the UI part alone can be quite a rabbithole :sweat_smile: 

Hope this helps somewhat, if not, feel free to get back to us. Thanks!

**Comment by pmig** [comment]
We currently don't support any kind of third party authentication besides using your existing `kubeconfig`. This is a blocker for exposing the UI publicly as the local kubeconfig is used from the user that starts the UI server.

We are considering adding this functionality into our Glasskube Cloud offering to connect your clusters and install and manage packages remotely, including a dedicated RBAC. So you can manage packages inside your cluster even without being connected to it or publicly exposing our UI. It will also help you with managing packages across multiple clusters.
